ftp-source/src/main/java/org/springframework/cloud/stream/module/ftp/FtpSourceProperties.java

```diff
@@ -15,118 +15,14 @@
 
 package org.springframework.cloud.stream.module.ftp;
 
-import java.io.File;
-
 import org.springframework.boot.context.properties.ConfigurationProperties;
+import org.springframework.cloud.stream.module.file.remote.AbstractRemoteFileSourceProperties;
 
 /**
  * @author David Turanski
+ * @author Gary Russell
  */
 @ConfigurationProperties
-public class FtpSourceProperties {
-
-
-	/**
-	 * The remote FTP directory, "/" by default.
-	 */
-	private String remoteDir = "/";
-
-	/**
-	 * Set to true to delete remote files after successful transfer, false by default.
-	 */
-	private boolean deleteRemoteFiles = false;
-
-	/**
-	 * The local directory to use for file transfers, System.getProperty("java.io.tmpdir") + "/xd/ftp" by default.
-	 */
-	private File localDir = new File(System.getProperty("java.io.tmpdir") + "/xd/ftp");
-
-	/**
-	 * Set to true to create the local directory if it does not exist, true by default
-	 */
-	private boolean autoCreateLocalDir = true;
-
-	/**
-	 * The suffix to use while the transfer is in progress, ".tmp" by default.
-	 */
-	private String tmpFileSuffix = ".tmp";
-
-	/**
-	 * A filter pattern to match the names of files to transfer, "*" by default.
-	 */
-	private String filenamePattern = "*";
-
-	/**
-	 * The remote file separator, "/" by default.
-	 */
-	private String remoteFileSeparator = "/";
-
-	/**
-	 * Set to true to preserve the original timestamp.
-	 */
-	private boolean preserveTimestamp = true;
-
-	public boolean isAutoCreateLocalDir() {
-		return autoCreateLocalDir;
-	}
-
-	public void setAutoCreateLocalDir(boolean autoCreateLocalDir) {
-		this.autoCreateLocalDir = autoCreateLocalDir;
-	}
-
-	public String getRemoteDir() {
-		return remoteDir;
-	}
-
-	public void setRemoteDir(String remoteDir) {
-		this.remoteDir = remoteDir;
-	}
-
-	public boolean isDeleteRemoteFiles() {
-		return deleteRemoteFiles;
-	}
-
-	public void setDeleteRemoteFiles(boolean deleteRemoteFiles) {
-		this.deleteRemoteFiles = deleteRemoteFiles;
-	}
-
-	public File getLocalDir() {
-		return localDir;
-	}
-
-	public void setLocalDir(File localDir) {
-		this.localDir = localDir;
-	}
-
-	public String getTmpFileSuffix() {
-		return tmpFileSuffix;
-	}
-
-	public void setTmpFileSuffix(String tmpFileSuffix) {
-		this.tmpFileSuffix = tmpFileSuffix;
-	}
-
-	public String getFilenamePattern() {
-		return filenamePattern;
-	}
-
-	public void setFilenamePattern(String filenamePattern) {
-		this.filenamePattern = filenamePattern;
-	}
-
-	public String getRemoteFileSeparator() {
-		return remoteFileSeparator;
-	}
-
-	public void setRemoteFileSeparator(String remoteFileSeparator) {
-		this.remoteFileSeparator = remoteFileSeparator;
-	}
-
-	public boolean isPreserveTimestamp() {
-		return preserveTimestamp;
-	}
+public class FtpSourceProperties extends AbstractRemoteFileSourceProperties {
```

Comment:
Maybe this is a way to address my earlier comment about port 21. Have dedicated subclasses?